src/compiler/csharp_plugin.cc

```diff
@@ -48,7 +48,29 @@ class CSharpGrpcGenerator : public grpc::protobuf::compiler::CodeGenerator {
                 const grpc::string &parameter,
                 grpc::protobuf::compiler::GeneratorContext *context,
                 grpc::string *error) const {
-    grpc::string code = grpc_csharp_generator::GetServices(file);
+    std::vector<std::pair<grpc::string, grpc::string> > options;
+    grpc::protobuf::compiler::ParseGeneratorParameter(parameter, &options);
+
+    bool generate_client = true;
```

Comment:
In general, I would recommend moving all of these into a "generator options" class that would have the proper defaults and everything, which makes it easier for maintenance in the future. Doesn't need to be done now however.